ext/drpttracer/drpttracer.h

```diff
@@ -0,0 +1,265 @@
+     * time.
+     * \note time_zero = perf_event_mmap_page.time_zero
+     */
+    uint64_t time_zero;
+} END_PACKED_STRUCTURE pt_metadata_t;
+
+/**
+ * The storage container type of a drpttracer's output.
+ * This data struct is used by drpttracer to store PT metadata, PT trace, and
+ * sideband data. These data can be dumped into different files. These files can be the
+ * inputs of drpt2ir, which decodes the PT data into Dynamorio's IR.
+ */
+typedef struct _drpttracer_output_t {
+    pt_metadata_t metadata;   /**< The PT trace's metadata. */
+    void *pt_buf;             /**< The PT trace's buffer pointer. */
+    size_t pt_buf_size;       /**< The buffer size of PT trace. */
+    void *sideband_buf;       /**< The PT sideband data's buffer pointer. */
+    size_t sideband_buf_size; /**< The buffer size of PT sideband data. */
+} drpttracer_output_t;
+
+/***************************************************************************
+ * INIT
+ */
+
+DR_EXPORT
+/**
+ * Initializes the drpttracer extension.  Must be called prior to any of the
+ * other routines.  Can be called multiple times (by separate components,
+ * normally) but each call must be paired with a corresponding call to
+ * drpttracer_exit().
+ *
+ * \return whether successful.
+ */
+bool
+drpttracer_init(void);
+
+DR_EXPORT
+/**
+ * Cleans up the drpttracer extension.
+ */
+void
+drpttracer_exit(void);
+
+/***************************************************************************
+ * PT TRACING APIS
+ */
+
+/**
+ * Success code for each drpttracer operation.
+ */
+typedef enum {
+    /** Operation succeeded. */
+    DRPTTRACER_SUCCESS,
+    /** Operation failed. */
+    DRPTTRACER_ERROR,
+    /** Operation failed: invalid tracing handle. */
+    DRPTTRACER_ERROR_INVALID_TRACING_HANDLE,
+    /** Operation failed: invalid tracing mode. */
+    DRPTTRACER_ERROR_INVALID_TRACING_MODE,
+    /** Operation failed: invalid output. */
+    DRPTTRACER_ERROR_INVALID_OUTPUT,
+    /** Operation failed: failed to open perf event. */
+    DRPTTRACER_ERROR_FAILED_TO_OPEN_PERF_EVENT,
+    /** Operation failed: failed to create tracer handle. */
+    DRPTTRACER_ERROR_FAILED_TO_CREATE_PTTRACER_HANDLE,
+    /** Operation failed: failed to start tracing. */
+    DRPTTRACER_ERROR_FAILED_TO_START_TRACING,
+    /** Operation failed: failed to stop tracing. */
+    DRPTTRACER_ERROR_FAILED_TO_STOP_TRACING,
+    /** Operation failed: failed to read trace. */
+    DRPTTRACER_ERROR_FAILED_TO_READ_TRACE,
+    /** Operation failed: failed to read sideband data. */
+    DRPTTRACER_ERROR_FAILED_TO_READ_SIDEBAND_DATA
+} drpttracer_status_t;
+
+/**
+ * The tracing modes for drpttracer_start_tracing().
+ *
+ * XXX: The sideband data collected in DRPTTRACER_TRACING_ONLY_USER and
+ * DRPTTRACER_TRACING_USER_AND_KERNEL modes don't include the initialization mmap2 event
+ * records. So that it will cause the drpt2ir to fail to find the images, the user can not
+ * use the sideband converter mode.
```

Comment:
Mention explicitly that the user and user_and_kernel modes are not completely supported yet.

Reply:
Done.